src/main/java/sg/ncl/MainController.java

```diff
@@ -5016,4 +5020,184 @@ private TeamQuota extractTeamQuotaInfo(String responseBody) {
     private String getExperimentMessage(String expName, String teamName) {
         return "Experiment " + expName + " in team " + teamName;
     }
+
+    @RequestMapping(value="/add_member/{teamId}", method= RequestMethod.GET)
+    public String addMember(@PathVariable String teamId, Model model) {
+        model.addAttribute("addMemberForm", new AddMemberForm());
+        return "add_member";
+    }
+
+    @RequestMapping(value="/add_member/{teamId}", method= RequestMethod.POST)
+    public String addMember(@PathVariable String teamId, @Valid AddMemberForm addMemberForm,
+                            final RedirectAttributes redirectAttributes)  throws WebServiceRuntimeException {
+        log.info("Starting to adding members by email for team {}", teamId);
+
+        String emails[] = addMemberForm.getEmails().split("\\r?\\n");
+
+        JSONArray jsonArray = new JSONArray();
+        for (int i = 0; i< emails.length; i++){
+            if (!VALID_EMAIL_ADDRESS_REGEX.matcher(emails[i]).matches()) {
+                addMemberForm.setErrMsg("Email address is not valid");
+                redirectAttributes.addFlashAttribute(MESSAGE, "Email address is not valid");
+                return "redirect:add_member";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "redirect:add_member" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)